dbt/adapters/spark/python_submissions.py

```diff
@@ -0,0 +1,284 @@
+import base64
+import time
+import requests
+from typing import Any, Dict
+import uuid
+
+import dbt.exceptions
+
+DEFAULT_POLLING_INTERVAL = 3
+SUBMISSION_LANGUAGE = "python"
+DEFAULT_TIMEOUT = 60 * 60 * 24
+
+
+class BasePythonJobHelper:
+    def __init__(self, parsed_model, credentials):
+        self.check_credentials(credentials)
+        self.credentials = credentials
+        self.identifier = parsed_model["alias"]
+        self.schema = getattr(parsed_model, "schema", self.credentials.schema)
+        self.parsed_model = parsed_model
+        self.timeout = self.get_timeout()
+        self.polling_interval = DEFAULT_POLLING_INTERVAL
+
+    def get_timeout(self):
+        timeout = self.parsed_model["config"].get("timeout", DEFAULT_TIMEOUT)
```

Comment:
I think this is probably cleaner way to set a timeout in the end since it pulls it from config. Really just a placeholder now. And we don't really want user to call `submit_python_job`, see https://github.com/dbt-labs/dbt-core/issues/5596

Reply:
I am gonna just merge this, feel free to open an issue if we feel like other ways to do timeout is better!